src/Plugin/Action/SystemPageRedirect.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/**
+ * @file
+ * Contains Drupal\rules\Plugin\Action\SystemPageRedirect.
+ */
+
+namespace Drupal\rules\Plugin\Action;
+
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\rules\Core\RulesActionBase;
+use Drupal\Core\Routing\RedirectDestinationInterface;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+
+/**
+ * Provides "Page redirect" rules action.
+ *
+ * @Action(
+ *   id = "rules_page_redirect",
+ *   label = @Translation("Page redirect"),
+ *   category = @Translation("System"),
+ *   context = {
+ *     "url" = @ContextDefinition("uri",
+ *       label = @Translation("URL"),
+ *       description = @Translation("A Drupal path, path alias, or external URL to redirect to. Enter (optional) queries after ? and (optional) anchor after #."),
+ *     ),
+ *     "force" = @ContextDefinition("boolean",
+ *       label = @Translation("Force redirect"),
+ *       description = @Translation("Force the redirect even if another destination parameter is present. Per default Drupal would redirect to the path given as destination parameter, in case it is set. Usually the destination parameter is set by appending it to the URL."),
+ *     ),
+ *     "destination" = @ContextDefinition("boolean",
+ *       label = @Translation("Append destination parameter"),
+ *       description = @Translation("Whether to append a destination parameter to the URL, so another redirect issued later on would lead back to the origin page."),
+ *     ),
+ *   },
+ *   provides = {
```

Comment:
Okay, I googled around and bothered a couple of Finns in DrupalcampCS in order to figure out how to do this Response listener thing. The nearest thing to what we need was http://cgit.drupalcode.org/globalredirect/tree/src/EventSubscriber/GlobalredirectSubscriber.php?h=8.x-1.x.

I got stuck with one problem and one question.

The problem: if we create event subscriber class which implements EventSubscriberInterface and handles redirections and add it through rules.services.yml, how will it know when was the rule action executed?

The question is simpler -- is this an overkill? Or is this redirection thing responsibility of core?